pkgs/applications/science/electronics/alliance/default.nix

```
@@ -1,74 +1,46 @@
 }:
 
 stdenv.mkDerivation rec {
   pname = "alliance";
-  version = "5.1.1";
-
-  src = fetchurl {
-    url = "http://www-asim.lip6.fr/pub/alliance/distribution/5.0/${pname}-${version}.tar.bz2";
-    sha256 = "046c9qwl1vbww0ljm4xyxf5jpz9nq62b2q0wdz9xjimgh4c207w1";
+  version = "unstable-2020-12-04";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.lip6.fr";
+    owner = "vlsi-eda";
+    repo = "alliance";
+    rev = "c12f377989571c1d868bebebdfb1574572e2ccf9";
+    sha256 = "RNfW621hRx5pzg5GhtY4sG8jwuQws3oG8EjZF7Mt5B0=";
   };
 
+  prePatch = "cd alliance/src";
 
   nativeBuildInputs = [ libtool automake autoconf flex ];
   buildInputs = [ xorgproto motif libX11 libXt libXpm bison ];
 
-  sourceRoot = "alliance/src/";
+  ALLIANCE_TOP = placeholder "out";
 
   configureFlags = [
-    "--prefix=$(out)"
+    "--prefix=${placeholder "out"}" "--enable-alc-shared"
   ];
 
-  preConfigure = ''
-    mkdir -p $out/etc
-
-    #texlive for docs seems extreme
-    mkdir -p $out/share/alliance
-    mv ./documentation $out/share/alliance
+  postPatch = ''
+    # texlive for docs seems extreme
     substituteInPlace autostuff \
-      --replace "$newdirs documentation" "$newdirs" \
-      --replace documentation Solaris
-
-    substituteInPlace sea/src/DEF_grammar_lex.l \
-      --replace "ifndef FLEX_BETA" "if (YY_FLEX_MAJOR_VERSION <= 2) && (YY_FLEX_MINOR_VERSION < 6)"
-    ./autostuff
-  '';
+      --replace "$newdirs documentation" "$newdirs"
 
-  allianceInstaller = ''
-    #!${runtimeShell}
-    cp -v -r -n --no-preserve=mode  $out/etc/* /etc/ > /etc/alliance-install.log
-  '';
+    substituteInPlace sea/src/DEF_grammar_lex.l --replace "ifndef FLEX_BETA" \
+      "if (YY_FLEX_MAJOR_VERSION <= 2) && (YY_FLEX_MINOR_VERSION < 6)"
 
-  allianceUnInstaller = ''
-    #!${runtimeShell}
-    awk '{print \$3}' /etc/alliance-install.log | xargs rm
-    awk '{print \$3}' /etc/alliance-install.log | xargs rmdir
-    rm /etc/alliance-install.log
-  '';
-
-  postInstall = ''
-    sed -i "s|ALLIANCE_TOP|$out|" distrib/*.desktop
-    mkdir -p $out/share/applications
-    cp -p distrib/*.desktop $out/share/applications/
-    mkdir -p $out/icons/hicolor/48x48/apps/
-    cp -p distrib/*.png $out/icons/hicolor/48x48/apps/
-
-    echo "${allianceInstaller}" > $out/bin/alliance-install
-    chmod +x $out/bin/alliance-install
-
-    echo "${allianceUnInstaller}" > $out/bin/alliance-uninstall
-    chmod +x $out/bin/alliance-uninstall
+    ./autostuff
   '';
 
   meta = with lib; {
-    description = "Complete set of free CAD tools and portable libraries for VLSI design";
-    homepage = "http://www-asim.lip6.fr/recherche/alliance/";
+    description = "(deprecated) Complete set of free CAD tools and portable libraries for VLSI design";
+    homepage = "http://coriolis.lip6.fr/";
     license = with licenses; gpl2Plus;
-    maintainers = with maintainers; [ ];
+    maintainers = with maintainers; [ l-as ];
     platforms = with platforms; linux;
```

does it have support for darwin? or is it just to broken to be patched
```suggestion
    platforms = platforms.linux;
```

---

I don't think it has any support for Darwin.